src/include/sof/audio/buffer.h

```diff
@@ -294,4 +294,20 @@ static inline int buffer_set_params(struct comp_buffer *buffer,
 	return 0;
 }
 
+/*
+ * This should be used for pipeline params negotiation only,
+ * where the component has more than 1 input or output, and
+ * the component itself only doing channel muxing/demuxing
+ * only so no frame_fmt or rate varies among buffers connected
+ * to it.
+ */
+static inline bool buffer_params_match(struct comp_buffer *buffer,
+				       struct sof_ipc_stream_params *params)
+{
+	assert(params && buffer);
+
+	return buffer->stream.frame_fmt == params->frame_fmt &&
+		buffer->stream.rate == params->rate;
+}
```

Comment:
Btw, what about channels ?

Reply:
the channels can be changed among different src/sink buffers of this kind of components(like demux, smart_amp, ...), so what we are negotiating here is frame_fmt and rate only.

Reply:
but SRC can change rate and pcm converter can change format. This does not scale.

Reply:
> but SRC can change rate and pcm converter can change format. This does not scale.

I actually not clear about how should it work once SRC or pcm converter added, looks there is no need to negotiate between pipelines anymore since SRC and pcm converter will help to do conversion for any supported format/rate. From this angle, you are right, this does not scale, we might need to remove this once the intelligent conversion is implemented.

Reply:
@lgirdwood since those converters are for long-term (may still with some opens) and no specific schedule for it yet, can we keep my version to fix the issue first?

Reply:
@keyonjie this API si still very tightly bound to a single use case i.e. muxing/demuxing. It would be far better to add a mask of the checks the user wants to validate
```c
buffer_params_match(struct comp_buffer *buffer,
				       struct sof_ipc_stream_params *params, uint32_t params_mask)
```
This would then lets users call it as follows
```c
/* validate frame format and rate match */
buffer_params_match(buffer, params, FRAME_FMT | RATE);
```